rados/conn.go

```diff
@@ -326,3 +326,53 @@ func (c *Conn) monCommand(args, inputBuffer []byte) (buffer []byte, info string,
 
 	return
 }
+
+// PGCommand sends a command to one of the PGs
+func (c *Conn) PGCommand(pgid, args []byte) (buffer []byte, info string, err error) {
+	return c.pgCommand(pgid, args, nil)
+}
+
+// PGCommand sends a command to one of the PGs, with an input buffer
+func (c *Conn) PGCommandWithInputBuffer(pgid, args, inputBuffer []byte) (buffer []byte, info string, err error) {
+	return c.pgCommand(pgid, args, inputBuffer)
+}
+
+func (c *Conn) pgCommand(pgid, args, inputBuffer []byte) (buffer []byte, info string, err error) {
+	name := C.CString(string(pgid))
+	argv := C.CString(string(args))
+	defer C.free(unsafe.Pointer(name))
+	defer C.free(unsafe.Pointer(argv))
+
+	var (
+		outs, outbuf       *C.char
+		outslen, outbuflen C.size_t
+	)
+	inbuf := C.CString(string(inputBuffer))
+	inbufLen := len(inputBuffer)
+	defer C.free(unsafe.Pointer(inbuf))
+
+	ret := C.rados_pg_command(c.cluster,
+		name,
+		&argv, 1,
```

Comment:
While I am new to these api's, a hardcoded '1' value here doesn't strike me as correct. Do you agree?

Reply:
I double checked https://github.com/ceph/ceph/blob/mimic/src/include/rados/librados.h#L3821. So, the `1` means that there is a single element in the parameter `argv`, which actually contains a single C string by `C.CString(string(args))`.

Reply:
Ah, I missed that the code was doing this. This also strikes me as incorrect. Compare to the [python bindings in the ceph tree](https://github.com/ceph/ceph/blob/master/src/pybind/rados/rados.pyx#L1478)  In that code args is a list that gets converted to an array of c strings. We need to mimic that behavior. I think the only reason this even compiles is that C does not distinguish a pointer-to-a-pointer from an array of pointers.
This also indicates that to more closely match the C and python bindings we want the function to accept a [][]byte for args.

Reply:
Sorry for the late response. I agree that to more closely match the C and Python APIs it should take `[][]byte`. But I checked the other equivalent API [MonCommand](https://github.com/ceph/go-ceph/blob/281fccac99ebb6c8b24c169c39c59d212a3df61c/rados/conn.go#L284) it only takes `[]byte` while it should take `[][]byte` as well to be consistent with C API https://github.com/ceph/ceph/blob/b6aa85cfb18f50dc9b27882c88a3d06f2ae6d1c5/src/pybind/rados/rados.pyx#L1330.

Reply:
That's a good point. Generally, I agree that internal consistency is good but I also don't like copying a mistake just to keep consistent. I would argue that we also need to update and/or deprecate that part of the api. That's not something I'm asking you to tackle, but we should make new additions like yours correct and then come back later and deal with the issues in the existing code (and there are plenty I've found so far in the project).

Reply:
Your statement totally makes sense to me. I will make changes for `PGCommand` at this time as you suggested.